src/components/triggersAndActions.ts

```diff
@@ -78,55 +89,103 @@ export const loadScheduledActions = async function (
 }
 
 export async function processTrigger(payload: TriggerPayload) {
+  const { id: trigger_id, trigger, table, record_id } = payload
+
+  // Deduce template ID -- different for each triggered table
+  const templateId = await DBConnect.getTemplateId(table, record_id)
+
   // Get Actions from matching Template
-  const result = await PostgresDB.getActionPluginsByTemplate(payload.table, {
-    record_id: payload.record_id,
-    trigger: payload.trigger,
-  })
+  const result = await DBConnect.getActionsByTemplateId(templateId, trigger)
+
   // Filter out Actions that don't match the current condition
-  const actions: ActionInTemplate[] = []
+  // and separate into Sequential and Async actions
+  const actionsSequential: ActionSequential[] = []
+  const actionsAsync: ActionInTemplate[] = []
 
   for (const action of result) {
     const condition = await evaluateExpression(action.condition, {
       objects: [payload],
-      pgConnection: PostgresDB,
+      pgConnection: DBConnect,
     })
-    if (condition) actions.push(action)
+    if (condition) {
+      if (action.sequence) actionsSequential.push(action as ActionSequential)
```

Comment:
can avoid extra code with different arrays by just checking for sequence for status, in loop below

Reply:
Yes, done. :)